### PR TITLE
drivers: udc: add USB device controller driver skeleton 

### PR DESCRIPTION
**Pull request by jfischer-no**
Add a USB device controller driver skeleton to use as a starting point
for implementing a specific driver.

### CODE REVIEW

File: drivers/usb/udc/udc_skeleton.c [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+		 */
+		LOG_DBG("ep 0x%02x halted", cfg->addr);
+		return 0;
+	}
+
+	return 0;
+}
+
+/*
+ * This is called in the contex of udc_ep_dequeue()
```
Comment (tmon-nordic) [inline drivers/usb/udc/udc_skeleton.c:108]: contex -> context

File: drivers/usb/udc/udc_skeleton.c [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+	}
+
+	irq_unlock(lock_key);
+
+	return 0;
+}
+
+/*
+ * Configure and make an endpoint ready for use.
+ * This is called in the contex of udc_ep_enable() or udc_ep_enable_internal(),
```
Comment (tmon-nordic) [inline drivers/usb/udc/udc_skeleton.c:134]: contex -> context

File: drivers/usb/udc/udc_skeleton.c [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+ */
+static int udc_skeleton_ep_disable(const struct device *dev,
+				   struct udc_ep_config *const cfg)
+{
+	LOG_DBG("Disable ep 0x%02x", cfg->addr);
+
+	return 0;
+}
+
+/* Halt endpoint. Halted enpoint should respond with a STALL handshake. */
```
Comment (tmon-nordic) [inline drivers/usb/udc/udc_skeleton.c:157]: enpoint -> endpoint

File: drivers/usb/udc/udc_skeleton.c [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+/*
+ * A UDC driver should always be implemented as a multi-instance
+ * driver, even if your platform does not require it.
+ */
+#define UDC_SKELETON_DEVICE_DEFINE(n)						\
+	K_THREAD_STACK_DEFINE(udc_skeleton_stack_##n, CONFIG_UDC_SKELETON);	\
+										\
+	static void udc_skeleton_thread_##n(void *dev, void *arg1, void *arg2)	\
+	{									\
+		while (1) {							\
```
Comment (tmon-nordic) [inline drivers/usb/udc/udc_skeleton.c:370]: There are two while loops now. The one here should probably be removed.

**Review by tmon-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**